yarn-project/noir-compiler/src/index.ts

```diff
@@ -41,8 +42,27 @@ export async function compileUsingNoirWasm(
   opts: NoirWasmCompileOptions,
 ): Promise<(ContractArtifact | ProgramArtifact)[]> {
   const cacheRoot = process.env.XDG_CACHE_HOME ?? join(process.env.HOME ?? '', '.cache');
-  const fileManager = new FileManager(fs, join(cacheRoot, 'aztec-noir-compiler'));
-  const compiler = NoirWasmContractCompiler.new(fileManager, resolve(projectPath), opts);
+  const fileManager = new FileManager(
+    {
+      ...fs,
+      ...{
+        existsSync,
+        mkdir: async (
+          dir: string,
+          opts?: {
+            /**
+             * Traverse child directories
+             */
+            recursive: boolean;
+          },
+        ) => {
+          await fs.mkdir(dir, opts);
+        },
+      },
+    },
```

Comment:
Does TS not accept `fs.mkdir` directly? Maybe we can make this an importable object?

Reply:
Nope, because of method overloaded types ): I extracted it to an importable function same as the memfs one